File: guides/java/kora-java-guide-dependency-injection-introduction-app/src/main/java/io/koraframework/guide/dependencyinjection/SmsNotifier.java
package io.koraframework.guide.dependencyinjection;

import io.koraframework.common.annotation.Component;
import io.koraframework.common.annotation.Tag;

@Tag(SmsTag.class)
@Component
public final class SmsNotifier implements Notifier {

    private final MessageFormatter formatter;

    public SmsNotifier(MessageFormatter formatter) {
        this.formatter = formatter;
    }

    @Override
    public String channel() {
        return "sms";
    }

    @Override
    public String notifyUser(String message) {
        return "SMS: " + formatter.format(message);
    }
}

File: guides/kotlin/kora-kotlin-guide-dependency-injection-introduction-app/src/main/kotlin/io/koraframework/guide/dependencyinjection/SmsNotifier.kt
package io.koraframework.guide.dependencyinjection

import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Tag

@Tag(SmsTag::class)
@Component
class SmsNotifier(
    private val formatter: MessageFormatter
) : Notifier {

    override fun channel(): String = "sms"

    override fun notifyUser(message: String): String = "SMS: ${formatter.format(message)}"
}
